core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -56,26 +56,66 @@
     /**
      * Constructor for simple inline script. The script will have no lang or
      * params set.
-     * 
+     *
      * @param script
      *            The inline script to execute.
      */
     public Script(String script) {
-        if (script == null) {
-            throw new IllegalArgumentException("The parameter script (String) must not be null in Script.");
-        }
-        this.script = script;
+        this(script, ScriptType.INLINE, null, null);
     }
 
     /**
      * For sub-classes to use to override the default language
      */
-    protected Script(String script, String lang) {
-        if (script == null) {
-            throw new IllegalArgumentException("The parameter script (String) must not be null in Script.");
-        }
-        this.script = script;
-        this.lang = lang;
+    public Script(String script, String lang) {
+        this(script, ScriptType.INLINE, lang, null);
+    }
+
+    /**
+     * Constructor for Script.
+     *
+     * @param script
+     *            The cache key of the script to be compiled/executed. For
+     *            inline scripts this is the actual script source code. For
+     *            indexed scripts this is the id used in the request. For on
+     *            file scripts this is the file name.
+     * @param lang
+     *            The language of the script to be compiled/executed.
+     * @param params
+     *            The map of parameters the script will be executed with.
+     */
+    public Script(String script, @Nullable String lang, @Nullable Map<String, ? extends Object> params) {
+        this(script, ScriptType.INLINE, lang, params);
+    }
+
+    /**
+     * Constructor for Script.
+     *
+     * @param script
+     *            The cache key of the script to be compiled/executed. For
+     *            inline scripts this is the actual script source code. For
+     *            indexed scripts this is the id used in the request. For on
+     *            file scripts this is the file name.
+     * @param type
+     *            The type of script -- dynamic, indexed, or file.
+     */
+    public Script(String script, ScriptType type) {
+        this(script, ScriptType.INLINE, null, null);
```

Comment:
Do we really need all these additional ctors? This seems to be a bug: `type` is not used here, but instead INLINE is always passed?